release.py

```diff
@@ -0,0 +1,498 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Release automation script."""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+import time
+import webbrowser
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import github
+import github.GitRelease
+import github.Repository
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class MyHandler(logging.StreamHandler[Any]):
+    """Custom logging handler to format messages based on their level."""
+
+    def format(self, record: logging.LogRecord):
+        """Format the log record based on its level."""
+        if record.levelno == logging.INFO:
+            self.setFormatter(logging.Formatter('ℹ️ INFO: %(message)s'))  # noqa: RUF001
+        elif record.levelno == logging.WARNING:
+            self.setFormatter(logging.Formatter('⚠️ WARNING: %(message)s'))
+        elif record.levelno == logging.ERROR:
+            self.setFormatter(logging.Formatter('❌ ERROR: %(message)s'))
+        return super().format(record)
+
+
+logger.addHandler(MyHandler())
+
+if 'GITHUB_TOKEN' not in os.environ:
+    logger.critical('Environment variable GITHUB_TOKEN not set.')
+    exit(1)
+
+OPS_VERSION_STR = r'version: str = \'(\d+\.\d+\.\d+(?:\.dev\d+)?)\''
```

Comment:
I would recommend start and end of line markers to be more precise.

```suggestion
OPS_VERSION_STR = r'^version: str = \'(\d+\.\d+\.\d+(?:\.dev\d+)?)\'$'
```

`re.xxx` calls may need the `re.MULTILINE` flag.

Without these markers, other matches are possible, for example:

```py
testing/src/scenario/_runtime.py
62:        juju_version: str = '3.0.0',
```